docs/architecture/airbyte-specification.md

```diff
@@ -0,0 +1,180 @@
+# The Airbyte Specification
+
+## Summary
+This page will describe the Airbyte Specification.
+
+####Contents:
+1. [General information about the specification](#General)
+1. [Integration primitives](#Primitives)
+1. [Details of the protocol to pass information between integrations](#The-Protocol)
+
+## General
+* All structs described in this article are defined using JsonSchema.
+* Airbyte uses json serialized representations of these structs for all inter-process communication.
+
```

Comment:
Not sure exactly where this belongs, but this should ideally have a one-liner saying ~`for a clearer idea of how integration images interact with the core Airbyte modules, see the architecture section` . basically something that helps motivate how this is used.

Reply:
good call. added it in general.